packages/genesys-spark-components/src/components/stable/gux-listbox/option-group/gux-option-group.tsx

```diff
@@ -9,24 +12,78 @@ import { Component, h, Host, JSX, Prop } from '@stencil/core';
   shadow: true
 })
 export class GuxOptionGroup {
+  private rootParent: HTMLElement;
+  private parentObserver: MutationObserver;
+
+  @Element() root: HTMLGuxOptionGroupBetaElement;
+
   @Prop()
   label!: string;
 
   // @Prop()
   // disabled: boolean;
   // TOOD: Add disabled state once design work is complete
 
+  @State()
+  filtered: boolean = false;
+
+  @State()
+  showDivider: boolean = true;
+
+  componentDidLoad(): void {
+    this.rootParent = getClosestElement(
+      'gux-dropdown',
+      this.root
+    ) as HTMLElement;
+
+    this.parentObserver = onMutation(this.rootParent, () => {
+      const groupOptions = Array.from(this.root?.children).filter(child =>
+        this.isOption(child)
+      ) as ListboxOptionElement[];
+      const allOptionsFiltered = groupOptions.every(option => option.filtered);
```

Comment:
This could be simplified a bit
```
const visibleOptions = Array.from(this.root?.children).filter(child =>
        this.isOption(child) && !child.filtered
      ) as ListboxOptionElement[];
```

Reply:
updated